test/e2e/storage/drivers/csi.go

```diff
@@ -0,0 +1,281 @@
+	o := utils.PatchCSIOptions{
+		OldDriverName:            h.driverInfo.Name,
+		NewDriverName:            h.driverInfo.Name + h.driverInfo.Framework.UniqueName,
+		DriverContainerName:      "hostpath",
+		ProvisionerContainerName: "csi-provisioner",
+		NodeName:                 h.driverInfo.Config.ServerNodeName,
+	}
+	cleanup, err := h.driverInfo.Framework.CreateFromManifests(func(item interface{}) error {
+		return utils.PatchCSIDeployment(h.driverInfo.Framework, o, item)
+	},
+		"test/e2e/testing-manifests/storage-csi/driver-registrar/rbac.yaml",
+		"test/e2e/testing-manifests/storage-csi/external-attacher/rbac.yaml",
+		"test/e2e/testing-manifests/storage-csi/external-provisioner/rbac.yaml",
+		"test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpath-attacher.yaml",
+		"test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpath-provisioner.yaml",
+		"test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml",
+		"test/e2e/testing-manifests/storage-csi/hostpath/hostpath/e2e-test-rbac.yaml",
+	)
+	h.cleanup = cleanup
+	if err != nil {
+		framework.Failf("deploying csi hostpath driver: %v", err)
+	}
+}
+
+func (h *hostpathCSIDriver) CleanupDriver() {
+	if h.cleanup != nil {
+		By("uninstalling csi hostpath driver")
+		h.cleanup()
+	}
+}
+
+// gce-pd
+type gcePDCSIDriver struct {
+	cleanup    func()
+	driverInfo DriverInfo
+}
+
+var _ TestDriver = &gcePDCSIDriver{}
+var _ DynamicPVTestDriver = &gcePDCSIDriver{}
+
+// InitGcePDCSIDriver returns gcePDCSIDriver that implements TestDriver interface
+func InitGcePDCSIDriver() TestDriver {
+	return &gcePDCSIDriver{
+		driverInfo: DriverInfo{
+			Name:        "com.google.csi.gcepd",
+			FeatureTag:  "[Serial]",
+			MaxFileSize: testpatterns.FileSizeMedium,
+			SupportedFsType: sets.NewString(
+				"", // Default fsType
+				"ext2",
+				"ext3",
+				"ext4",
+				"xfs",
+			),
+			IsPersistent:       true,
+			IsFsGroupSupported: true,
+			IsBlockSupported:   false,
+		},
+	}
+}
+
+func (g *gcePDCSIDriver) GetDriverInfo() *DriverInfo {
+	return &g.driverInfo
+}
+
+func (g *gcePDCSIDriver) SkipUnsupportedTest(pattern testpatterns.TestPattern) {
+	f := g.driverInfo.Framework
+	cs := f.ClientSet
+	config := g.driverInfo.Config
+	framework.SkipUnlessProviderIs("gce", "gke")
+	framework.SkipIfMultizone(cs)
+
+	// TODO(#62561): Use credentials through external pod identity when that goes GA instead of downloading keys.
+	createGCESecrets(cs, config)
+	framework.SkipUnlessSecretExistsAfterWait(cs, "cloud-sa", config.Namespace, 3*time.Minute)
+}
+
+func (g *gcePDCSIDriver) GetDynamicProvisionStorageClass(fsType string) *storagev1.StorageClass {
+	ns := g.driverInfo.Framework.Namespace.Name
+	provisioner := g.driverInfo.Name
+	suffix := fmt.Sprintf("%s-sc", g.driverInfo.Name)
+
+	parameters := map[string]string{"type": "pd-standard"}
+
+	return getStorageClass(provisioner, parameters, nil, ns, suffix)
+}
```

Comment:
@mkimuram Should this gcePDCSIDriver.GetDynamicProvisionStorageClass really ignore the fsType string? The driver declares that it supports multiple different filesystems.

Does this lead to multiple different test configurations (one for each filesystem) where each individual test run then uses the same default fs?

Reply:
@pohly 

Thank you for pointing it out. Current ```gcePDCSIDriver.GetDynamicProvisionStorageClass``` code lacks setting ```fsType``` as [it was done for in-tree one](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/drivers/in_tree.go#L1176). Also, I came to think that volumes test should test if the volume is mounted with the proper fsType, if fsType is specified. (Currently, it [tests multiple fsType](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/volumes.go#L44) but doesn't seem to [test fsType](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/volume_util.go#L404).)

I will make a PR to fix it.

Reply:
Opened an issue and a PR below:

Issue: https://github.com/kubernetes/kubernetes/issues/70822
PR: https://github.com/kubernetes/kubernetes/pull/70823